yurt/yurt_core/add.py

```diff
@@ -104,32 +120,34 @@ def remote_server(**kwargs):
                       "the input you enter matches one of those choices.\n"
                       "Press Enter to Continue.")
 
-    # TODO: Abstract this question-loop with a re-usable util function
-    try:
-        question_items = ATTRIBUTE_TO_QUESTION_MAPPING.iteritems()
-    except AttributeError:
-        question_items = ATTRIBUTE_TO_QUESTION_MAPPING.items()
-
-    for attribute, prompt in question_items:
-        if kwargs[attribute] is None:
-            settings[attribute] = raw_input_wrapper(prompt, attribute in lowercase_attrs)
-            # Handle gunicorn defaults
-            if attribute == "num_gunicorn_workers" and settings[attribute] == "":
-                settings[attribute] = "2"
-            if attribute == "gunicorn_max_requests" and settings[attribute] == "":
-                settings[attribute] = "0"
-            # Handle email defaults
-            if attribute == "email_port" and settings[attribute] == "":
-                settings[attribute] = "687"
-            if attribute == "email_use_ssl" and settings[attribute] == "":
-                settings[attribute] = "False"
-            if attribute == "email_use_tls" and settings[attribute] == "":
-                settings[attribute] = "True"
-            # Handle other defaults
-            if attribute == "multiple_yurt_project_server" and settings[attribute] == "":
-                settings[attribute] = "no"
-        else:
-            settings[attribute] = kwargs[attribute]
+    def handle_defaults(result, attribute):
```

Comment:
is there a more clever way to do this? have a dict of default values then merge in the dict of user supplied ones to just overwrite existing values?